### PR TITLE
Reshape input to 2D for softmax, and allow 1 input for concat

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by Honry** [comment]
@huningxin, PTAL, thanks!

File: tensorflow/lite/delegates/webnn/webnn_delegate.cc [hunk 1]
```diff
@@ -1762,7 +1762,7 @@ class Subgraph {
       std::unordered_map<int, emscripten::val>& webnn_operands,
       const bool detect_supported_op) {
     TF_LITE_ENSURE_STATUS(
-        CheckNumInputsAndOutputs(logging_context, node, 2, 4, 1, node_index));
+        CheckNumInputsAndOutputs(logging_context, node, 1, 4, 1, node_index));
```
Comment (huningxin) [inline tensorflow/lite/delegates/webnn/webnn_delegate.cc]: Thanks for this fix. Better to have its own commit.

File: tensorflow/lite/delegates/webnn/webnn_delegate.cc [hunk 2]
```diff
@@ -2564,15 +2564,34 @@ class Subgraph {
...
                      webnn_operands.at(input_tensor_id).as<bool>());
-      webnn_operands.insert(
-          std::make_pair(output_tensor_id,
-                         builder.call<emscripten::val>(
-                             "softmax", webnn_operands.at(input_tensor_id))));
+      emscripten::val output = emscripten::val::object();
+      // WebNN Softmax only support 2d input shape, reshape input to 2d.
+      if (input_shape.size() != 2) {
+        emscripten::val new_shape = emscripten::val::array();
+        new_shape.call<void>("push", emscripten::val::null());
```
Comment (huningxin) [inline tensorflow/lite/delegates/webnn/webnn_delegate.cc:2581]: Dwayne proposed to remove the support of `null` in newShape. https://github.com/webmachinelearning/webnn/issues/388

You may want to add a TODO comment here, once it is in spec and implementation. We should compute the dimension size by the delegate.

File: tensorflow/lite/delegates/webnn/webnn_delegate.cc (hunk 1 above)
Comment (Honry) [reply tensorflow/lite/delegates/webnn/webnn_delegate.cc]: Done via https://github.com/huningxin/tensorflow/pull/41.

File: tensorflow/lite/delegates/webnn/webnn_delegate.cc (hunk 2 above)
Comment (Honry) [reply tensorflow/lite/delegates/webnn/webnn_delegate.cc:2581]: Thanks for the reminder, TODO added.

**Review by huningxin — APPROVED** [review]
